### ISSUE TITLE
A negative number behind a call is closed up into a subtraction under Sass and Less

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#264]
Found while fixing [#257](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/257), which taught `function-whitespace-after` to leave the whitespace that makes a sign the operator of a sum. That reading turns on what follows the sign: a digit makes it part of a number, anything else makes it an operator. In Sass and in Less a sign with a digit behind it is an operator all the same, and the whitespace in front of it is what says so.

### What minimal example or steps are needed to reproduce the bug?

A Sass value holding a call and a negative number:

```scss
a { b: foo($a) -2px; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-whitespace-after": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --custom-syntax postcss-scss --config myconfig.json --fix
```

### What did you expect to happen?

Nothing reported. Sass reads `foo($a) -2px` as a list of two values, and reads `foo($a)-2px` as a subtraction: the whitespace in front of the sign is the whole of what tells the two apart. Whichever of the two the author wrote, the fix must not write the other.

### What actually happened?

`1:15 Unexpected whitespace after ")"`, and the fix writes:

```scss
a { b: foo($a)-2px; }
```

A list has become an expression. Less goes the same way with `a { b: foo(@a) -2px; }`, whose `-` its own arithmetic reads.

### What is the cause?

`LEADING_SPACED_SUM_OPERATOR`, the name [#257](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/257) added, refuses a sign that opens a number: `/^[ \t\n\r\f]+[+-](?![\d.])/u`. That refusal is right for CSS — `calc(1px -1px)` is not a sum in any browser, `-1px` there being a single number token — and it is wrong for the two syntaxes that spell their own arithmetic, where a space in front of the sign is the operator and nothing else is.

The rule reads the same way whatever syntax spelled the file, and nothing in it asks.

### Do you have a proposal to fix the bug?

Ask which syntax spelled the file, as `searchCopy` already asks in order to know what a double slash opens, and drop the lookahead where the answer is `postcss-scss` or `postcss-less`. The whitespace in front of any sign is then left alone under those two, which is the reading both languages have.

What must not be done is dropping the lookahead everywhere: `a { b: url(x) -1px; }` is a list of two values in plain CSS whether the space stands there or not, and closing it up is the very thing this option is for.
